### ISSUE TITLE
Terraform wants to create aws_cloudwatch_metric_alarm on each run

### ISSUE THREAD

**Issue by looppair** [hashicorp/terraform#6550]
Every time I run a plan with terraform, it suggests creating an aws_cloudwatch_metric_alarm which terraform has already created. Additionally, the resource never appears in the terraform state file.

### Terraform Version
Terraform v0.6.15

### Affected Resource(s)
- aws_cloudwatch_metric_alarm

### Terraform Configuration Files
```hcl
resource "aws_cloudwatch_metric_alarm" "asg_alarm_lambda_alarm" {
  alarm_name = "${aws_lambda_function.asg_alarm.function_name}_lambda_alarm"
  comparison_operator = "GreaterThanThreshold"
  evaluation_periods = 1
  metric_name = "Errors"
  namespace = "AWS/Lambda"
  period = 60
  statistic = "Sum"
  threshold = 0
  alarm_actions = [
    "${aws_sns_topic.asg_alerts.arn}"]
  dimensions {
    FunctionName = "${aws_lambda_function.asg_alarm.function_name}"
  }
}
```

### Expected Behavior
Terraform should not have attempted to plan/create existing aws_cloudwatch_metric_alarm

### Actual Behavior

Amazon account code omitted.
```
+ aws_cloudwatch_metric_alarm.asg_alarm_lambda_alarm
    actions_enabled:          "" => "1"
    alarm_actions.#:          "" => "1"
    alarm_actions.3734395141: "" => "arn:aws:sns:us-east-1:xxxxxx:qa_asg_alerts"
    alarm_name:               "" => "qa_asg_alarm_lambda_alarm"
    comparison_operator:      "" => "GreaterThanThreshold"
    dimensions.#:             "" => "1"
    dimensions.FunctionName:  "" => "qa_asg_alarm"
    evaluation_periods:       "" => "1"
    metric_name:              "" => "Errors"
    namespace:                "" => "AWS/Lambda"
    period:                   "" => "60"
    statistic:                "" => "Sum"
    threshold:                "" => "0"


Plan: 1 to add, 0 to change, 0 to destroy.
```

**Comment by stack72** [comment]
Hi @looppair 

Thanks for the report, sorry this is causing issues. Please can you post the plan differences that you are seeing? This will help try and debug the issue

Thanks

Paul

**Comment by looppair** [comment]
Thanks for the response.  Updated above.

**Comment by stack72** [comment]
Hi @looppair 

I have just tried the following config:

```
resource "aws_s3_bucket" "lambda_bucket" {
  bucket = "tf-test-bucket-12434"
}

resource "aws_s3_bucket_object" "lambda_code" {
  bucket = "${aws_s3_bucket.lambda_bucket.id}"
  key = "lambdatest.zip"
  source = "lambdatest.zip"
}

resource "aws_iam_role" "iam_for_lambda" {
    name = "iam_for_lambda"
    assume_role_policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "sts:AssumeRole",
      "Principal": {
        "Service": "lambda.amazonaws.com"
      },
      "Effect": "Allow",
      "Sid": ""
    }
  ]
}
EOF
}

resource "aws_lambda_function" "asg_alarm" {
    s3_bucket = "${aws_s3_bucket.lambda_bucket.id}"
    s3_key = "${aws_s3_bucket_object.lambda_code.id}"
    function_name = "example_lambda_name_s3"
    role = "${aws_iam_role.iam_for_lambda.arn}"
    handler = "exports.example"
}

resource "aws_cloudwatch_metric_alarm" "asg_alarm_lambda_alarm" {
  alarm_name = "${aws_lambda_function.asg_alarm.function_name}_lambda_alarm"
  comparison_operator = "GreaterThanThreshold"
  evaluation_periods = 1
  metric_name = "Errors"
  namespace = "AWS/Lambda"
  period = 60
  statistic = "Sum"
  threshold = 0
  dimensions {
    FunctionName = "${aws_lambda_function.asg_alarm.function_name}"
  }
}
```

This works as expected for me. I understand there will be some differences between my code and yours. Can you post what happens on a terraform apply, then a terraform plan?

the code above only suggests that it's adding a cloud metric alarm for the first time. What is the output from the terraform apply?

P.

**Comment by looppair** [comment]
On terraform apply, I see:

```
aws_cloudwatch_metric_alarm.asg_alarm_lambda_alarm: Creating...
  actions_enabled:          "" => "1"
  alarm_actions.#:          "" => "1"
  alarm_actions.3734395141: "" => "arn:aws:sns:us-east-1:xxxxxx:qa_asg_alerts"
  alarm_name:               "" => "qa_asg_alarm_lambda_alarm"
  comparison_operator:      "" => "GreaterThanThreshold"
  dimensions.#:             "" => "1"
  dimensions.FunctionName:  "" => "qa_asg_alarm"
  evaluation_periods:       "" => "1"
  metric_name:              "" => "Errors"
  namespace:                "" => "AWS/Lambda"
  period:                   "" => "60"
  statistic:                "" => "Sum"
  threshold:                "" => "0"
aws_cloudwatch_metric_alarm.asg_alarm_lambda_alarm: Creation complete
```
Any subsequent plan produces the output mentioned in the OP Actual Behaviour.

**Comment by loopsysdev** [comment]
It turns out we were missing a cloudwatch:DescribeAlarms permission and this was unfortunately causing terraform to create a plan as if the alarm did not exist. Maybe terraform should report that it is missing the permission?

**Comment by dtolnay** [comment]
I fixed this in #7227.
